client-keycloak/src/main/java/org/apache/atlas/keycloak/client/KeycloakClient.java

```diff
@@ -0,0 +1,186 @@
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.atlas.ApplicationProperties.ATLAS_CONFIGURATION_DIRECTORY_PROPERTY;
+
+public class KeycloakClient {
+    public static final Logger LOG = LoggerFactory.getLogger(KeycloakClient.class);
+
+    private static KeycloakClient keycloakClient = null;
+    public static Keycloak keycloak = null;
+
+    public static final String KEYCLOAK_PROPERTIES = "keycloak.json";
+
+    private static String REALM_ID;
+    private static String AUTH_SERVER_URL;
+    private static String CLIENT_ID;
+    private static String CLIENT_SECRET;
+    private static String GRANT_TYPE;
+
+
+    private KeycloakClient() {
+    }
+
+    public static KeycloakClient getKeycloakClient() throws AtlasBaseException {
+        if (keycloakClient == null) {
+            LOG.info("Initializing Keycloak client..");
+            try {
+                initConf();
+            } catch (IOException e) {
+                LOG.error("Failed to fetch Keycloak conf {}", e.getMessage());
+                throw new AtlasBaseException(e);
+            } catch (JSONException e) {
+                LOG.error("Failed to parsing Keycloak conf {}", e.getMessage());
+                throw new AtlasBaseException(e);
+            } catch (Exception e) {
+                LOG.error("Failed to connect to Keycloak {}", e.getMessage());
+                throw new AtlasBaseException(e);
+            }
+            init();
+            LOG.info("Initialized Keycloak client..");
+        }
+
+        if (keycloak.isClosed()) {
+            LOG.info("Re-initializing keycloak client");
+            init();
+        }
+
+        return keycloakClient;
+    }
+
+    private static void initConf() throws Exception {
+        String confLocation = System.getProperty(ATLAS_CONFIGURATION_DIRECTORY_PROPERTY);
+
+        File confFile;
+        if (StringUtils.isNotEmpty(confLocation)) {
+            confFile = new File(confLocation, KEYCLOAK_PROPERTIES);
+
+            if (confFile.exists()) {
+                String keyConf = new String(Files.readAllBytes(confFile.toPath()), StandardCharsets.UTF_8);
+
+                JSONObject object = new JSONObject(keyConf);
+
+                REALM_ID = object.getString("realm");
+                AUTH_SERVER_URL = object.getString("auth-server-url");
+                CLIENT_ID = object.getString("resource");
+                GRANT_TYPE = "client_credentials";
+                CLIENT_SECRET = object.getJSONObject("credentials").getString("secret");
+
+                LOG.info("Keycloak conf: REALM_ID:{}, AUTH_SERVER_URL:{}, CLIENT_ID:{}, CLIENT_SECRET:{}",
+                        REALM_ID, AUTH_SERVER_URL, CLIENT_ID, CLIENT_SECRET);
+            } else {
+                throw new AtlasBaseException("Keycloak configuration file not found in location " + confLocation);
+            }
+        }
```

Comment:
do you think exception should be thrown if none of the conditions met ? I see we arent in case of `if (StringUtils.isNotEmpty(confLocation))`

Reply:
fixed in https://github.com/atlanhq/atlas-metastore/pull/1737